db/inmemory_test.go

```diff
@@ -0,0 +1,54 @@
+package db
+
+import "testing"
+
+func TestBasic(t *testing.T) {
+	im := NewInMemoryVolumeDatabase()
+
+	err := im.Connect()
+	if err != nil {
+		t.Fatal("Connect returned an error! " + err.Error())
+	}
+
+	list, err := im.List()
+	if err != nil {
+		t.Fatal("List returned an error! " + err.Error())
```

Comment:
(Side point: if we make a true test where we assume _nothing_ about the underlying code, it is good to test for corrupted state between function calls as well.)